### PR TITLE
add an extra arg to possibly also scan the corrected files

### PR DESCRIPTION
**Pull request by mgallien**
Signed-off-by: Matthieu Gallien <matthieu.gallien@nextcloud.com>

### CODE REVIEW

**Comment by artonge** [comment]
I thought about this, but I think that most cases if not all cases will not benefit from it as it will slow down the script way too much. So not a good idea to offer this feature. Do you have a request for it ?

**Comment by mgallien** [comment]
> I thought about this, but I think that most cases if not all cases will not benefit from it as it will slow down the script way too much. So not a good idea to offer this feature. Do you have a request for it ?

Imagine a case where the server has way many files but only a few ones are affected by the wrong date bug. I guess in that case we would benefit from this.
So I would like to be able to suggest that in cases where only a few files are affected.

**Comment by mgallien** [comment]
@artonge, if it is off by default I guess it will be better to be able to use it for cases where storage has many files but only `0.1%` are affected

File: solvable_files.sh [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
 
 export data_dir="$1"
 export db_type="$2"
 export db_host="$3"
 export db_user="$4"
 export db_pwd="$5"
 export db_table="$6"
-export action="${7:-list}"
+export action="$7"
+export scan_action="${8:-list}"
```
Comment (artonge) [inline solvable_files.sh]:
```suggestion
export action="${7:-list}"
export scan_action="${8:-noscan}"
```

File: solvable_files.sh [hunk 2]
```diff
@@ -24,6 +25,8 @@ fi
 # 4. Return if mtime_on_fs != mtime_in_db
 # 5. Correct the fs mtime with touch
 function correct_mtime() {
+	set -eu
```
Comment (artonge) [inline solvable_files.sh]: Need removal

File: solvable_files.sh [hunk 3]
```diff
@@ -118,6 +121,10 @@ function correct_mtime() {
 	if [ "$action" = "fix" ]
 	then
 		touch --no-create "$filepath"
+		if [ "$scan_action" = "scan" ]
+		then
+			sudo -u www-data php ./occ files:scan --quiet --path="${relative_filepath}"
```
Comment (artonge) [inline solvable_files.sh]: This introduces requirements on how to run occ, having sudo, using the www-data user and being in the nextcloud root directory. Let's go for that now, if someone needs more, we'll introduce a better way of passing arguments to let the user have more control over it.

```suggestion
			sudo -u www-data php ./occ files:scan --quiet --path="$relative_filepath"
```

**Review by artonge — CHANGES_REQUESTED** [review]

File: solvable_files.sh (hunk 3 above)
Comment (mgallien) [reply solvable_files.sh]: yes sorry forgot about that because I was in a hurry

File: solvable_files.sh (hunk 1 above)
Comment (mgallien) [reply solvable_files.sh]: learnt something
thanks

**Comment by mgallien** [comment]
@artonge is it good now ?
I will need it today (if you agree)

File: solvable_files.sh (hunk 3 above)
Comment (artonge) [reply solvable_files.sh]: Still need that one.

Comment (mgallien) [reply solvable_files.sh]: yes sorry just saw it after

Comment (mgallien) [reply solvable_files.sh]: @artonge  done, sorry

**Review by artonge — APPROVED** [review]

**Pull request merged**